python/lsst/summit/utils/utils.py

```diff
@@ -1419,3 +1425,221 @@ def getSunAngle(time: Time | None = None) -> float:
+
+
+@dataclass
+class RobustFitResult:
+    intercept: float
+    slope: float
+    scatter: float
+    slope_pvalue: float
+    slope_stderr: float
+    slope_tvalue: float
+    intercept_pvalue: float
+    intercept_stderr: float
+    intercept_tvalue: float
+
+
+class RobustFitter:
+    """
+    Robust linear fit using RANSAC,
+    with confidence and prediction intervals from statsmodels.
+    """
+
+    def __init__(
+        self,
+        x: np.ndarray,
+        y: np.ndarray,
+        min_samples: float = 0.2,
+        residual_threshold: float | None = None,
+        random_state: int = 42,
+    ):
+        self.min_samples = min_samples
+        self.residual_threshold = residual_threshold
+        self.random_state = random_state
+
+        if self.residual_threshold is None:
+            # Set a default residual threshold based on the data
+            if np.isnan(x).all() or np.isnan(y).all():
+                raise ValueError("Cannot fit model: all input data is NaN.")
+            self.residual_threshold = 1.5 * mad_std(y)
+
+        self.fit(x, y)
+
+    def fit(self, x, y):
+        # Ensure x and y are numpy arrays
+        x = np.asarray(x)
+        y = np.asarray(y)
+
+        # check for NaNs
+        is_nan = np.isnan(x) | np.isnan(y)
+        if np.all(is_nan):
+            raise ValueError("All input data is NaN. Cannot fit a model.")
+
+        X = np.asarray(x[~is_nan]).reshape(-1, 1)
+        yfit = np.asarray(y[~is_nan])
+
+        self.model = RANSACRegressor(
+            estimator=LinearRegression(),
+            min_samples=self.min_samples,
+            residual_threshold=self.residual_threshold,
+            random_state=self.random_state,
+        )
+        self.model.fit(X, yfit)
+        self.slope = self.model.estimator_.coef_[0]
+        self.intercept = self.model.estimator_.intercept_
+
+        # Flatten x to ensure it is 1D for statsmodels
+        self.x = x.flatten()
+        self.y = np.asarray(y)
+
+        # Use only inliers for statistics
+        self.outlier_mask = np.full(self.x.shape, False, dtype=bool)
+        self.outlier_mask[is_nan] = True  # Keep NaNs as outliers
+        # Mark outliers as True, inliers as False
+        self.outlier_mask[~is_nan] = ~self.model.inlier_mask_
+
+        x_inlier = self.x[~self.outlier_mask]
+        y_inlier = self.y[~self.outlier_mask]
+
+        # OLS with statsmodels for CI and p-value
+        X_design = sm.add_constant(x_inlier)
+        ols_model = sm.OLS(y_inlier, X_design).fit()
+
+        # statsmodels reports [intercept, slope]
+        ci = ols_model.conf_int(alpha=0.32)  # 68% CI
+        self.ci_intercept = ci[0]
+        self.ci_slope = ci[1]
+
+        self.slope_pvalue = ols_model.pvalues[1]
+        self.slope_stderr = ols_model.bse[1]
+        self.slope_tvalue = ols_model.tvalues[1]
+        self.scatter = np.std(ols_model.resid)
+        self.ols_model = ols_model  # Expose full model if needed
+
+    def report_best_values(self):
```

Comment:
why is this a separate call? Why not just return the result from the fit method?